src/shogun/evaluation/SigmoidCalibration.cpp

```diff
@@ -0,0 +1,188 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Heiko Strathmann, Dhruv Arya
+ */
+
+#include <shogun/evaluation/SigmoidCalibration.h>
+#include <shogun/mathematics/Math.h>
+#include <shogun/mathematics/Statistics.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+
+using namespace shogun;
+
+CSigmoidCalibration::CSigmoidCalibration() : CCalibration()
+{
+	init();
+}
+
+CSigmoidCalibration::~CSigmoidCalibration()
+{
+}
+
+void CSigmoidCalibration::init()
+{
+	m_maxiter = 100;
+	m_minstep = 1E-10;
+	m_sigma = 1E-12;
+	m_epsilon = 1E-5;
+
+	SG_ADD(&m_sigmoid_as, "m_sigmoid_as", "Vector of paramter A of sigmoid for each class.", MS_NOT_AVAILABLE);
+	SG_ADD(&m_sigmoid_bs, "m_sigmoid_bs", "Vector of paramter B of sigmoid for each class.", MS_NOT_AVAILABLE);
+	SG_ADD(&m_maxiter, "m_maxiter", "Maximum number of iteration for search.", MS_NOT_AVAILABLE);
+	SG_ADD(&m_minstep, "m_minstep", "Minimum step taken in line search.", MS_NOT_AVAILABLE);
+	SG_ADD(&m_sigma, "m_sigma", "Positive parameter to ensure positive semi-definite Hessian.", MS_NOT_AVAILABLE);
+	SG_ADD(&m_epsilon, "m_epsilon", "Stopping criteria.", MS_NOT_AVAILABLE);
+}
+
+void CSigmoidCalibration::set_maxiter(index_t maxiter)
+{
+	m_maxiter = maxiter;
+}
+
+index_t CSigmoidCalibration::get_maxiter()
+{
+	return m_maxiter;
+}
+
+void CSigmoidCalibration::set_minstep(float64_t minstep)
+{
+	m_minstep = minstep;
+}
+
+float64_t CSigmoidCalibration::get_minstep()
+{
+	return m_minstep;
+}
+
+void CSigmoidCalibration::set_sigma(float64_t sigma)
+{
+	m_sigma = sigma;
+}
+
+float64_t CSigmoidCalibration::get_sigma()
+{
+	return m_sigma;
+}
+
+void CSigmoidCalibration::set_epsilon(float64_t epsilon)
+{
+	m_epsilon = epsilon;
+}
+
+float64_t CSigmoidCalibration::get_epsilon()
+{
+	return m_epsilon;
+}
+
+bool CSigmoidCalibration::fit_binary(CBinaryLabels* predictions, CBinaryLabels* targets)
+{
+	m_sigmoid_as.resize_vector(1);
+	m_sigmoid_bs.resize_vector(1);
+	auto sigmoid_params = CStatistics::fit_sigmoid(predictions->get_values(), targets->get_labels(), 
+		m_maxiter, m_minstep, m_sigma, m_epsilon);
+
+	m_sigmoid_as[0] = sigmoid_params.a;
+	m_sigmoid_bs[0] = sigmoid_params.b;
+
+	return true;
+}
+
+CBinaryLabels*
+CSigmoidCalibration::calibrate_binary(CBinaryLabels* predictions)
+{
+	CStatistics::SigmoidParamters params; 
+	params.a = m_sigmoid_as[0];
+	params.b = m_sigmoid_bs[0];
+
+	/** Convert predictions to probabilties. */
+	auto values = calibrate_values(predictions->get_values(), params);
```

Comment:
rather than the comment, I would name the variable `probabilities` and then you dont need it. Minor of course

Reply:
That would look much better indeed